Makefile

```diff
@@ -8,268 +28,417 @@ usage:
 	@echo "          alldoc: Builds Tock documentation for all boards"
-	@echo "           audit: Audit Cargo dependencies for all kernel sources"
-	@echo "              ci: Run all continuous integration tests"
 	@echo "           clean: Clean all builds"
-	@echo "          clippy: Runs the clippy code linter with Tock's default arguments"
-	@echo " emulation-check: Run the emulation tests for supported boards"
-	@echo " emulation-setup: Setup QEMU for the emulation tests"
 	@echo "          format: Runs the rustfmt tool on all kernel sources"
-	@echo "    format-check: Checks if the rustfmt tool would require changes, but doesn't make them"
-	@echo "            list: Lists available boards"
+	@echo
+	@echo "The make system also drives all continuous integration and testing:"
+	@echo "         $$(tput bold)prepush$$(tput sgr0): Fast checks to run before pushing changes upstream"
+	@echo "          ci-all: Run all continuous integration tests (possibly slow!)"
+	@echo "         ci-help: More information on Tock CI and testing"
 	@echo
 	@echo "$$(tput bold)Happy Hacking!$$(tput sgr0)"
 
+##
+## End: usage.
+##
+################################################################################
+##
+## Utility functions
+##
+
+define banner
+	@printf "\n"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@string="$(1)" && printf "$$(tput bold)* %-$$((76))s *\n" "$$string"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@printf "\n"
+endef
+
+##
+## End: functions.
+##
+################################################################################
+##
+## User convenience targets
+##
+
+# Aggregate targets
+.PHONY: allaudit
+audit:
+	@for f in `./tools/list_lock.sh`;\
+		do echo "$$(tput bold)Auditing $$f";\
+		(cd "$$f" && cargo audit || exit 1);\
+		done
+
 .PHONY: allboards
 allboards:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Build $$f"; $(MAKE) -C "boards/$$f" || exit 1; done
+	@for f in $(ALL_BOARDS);\
+		do echo "$$(tput bold)Build $$f";\
+		$(MAKE) -C "boards/$$f" || exit 1;\
+		done
 
 .PHONY: allcheck
 allcheck:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Check $$f"; $(MAKE) -C "boards/$$f" check || exit 1; done
+	cargo check
 
 .PHONY: alldoc
 alldoc:
-	@for f in `./tools/list_boards.sh`; do echo "$$(tput bold)Documenting $$f"; $(MAKE) -C "boards/$$f" doc || exit 1; done
+	@for f in $(ALL_BOARDS);\
+		do echo "$$(tput bold)Documenting $$f";\
+		$(MAKE) -C "boards/$$f" doc || exit 1;\
+		done
 
 
+# Commands
+.PHONY: fmt format
+fmt format: tools/.format_fresh
+	$(call banner,Formatting complete)
 
-###################################################################
+# Get a list of all rust source files (everything fmt operates on)
+$(eval RUST_FILES_IN_TREE := $(shell git ls-files | grep '\.rs$$'))
```

Comment:
This breaks `make format` if you download the .zip file. It also causes this less-than-friendly message to print: `fatal: not a git repository (or any of the parent directories): .git`.

Reply:
Fixed.